hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -792,6 +830,244 @@ public Bytes getFailureResultFor(final ResponseCodeEnum status) {
 		}
 	}
 
+	private abstract class AbstractCreatePrecompile implements Precompile {
+		//TODO: if possible, use this class for all token create scenarios as is done in @TransferPrecompile
+
+		protected TokenCreateWrapper tokenCreateOp;
+
+		protected final TransactionBody.Builder createSyntheticTokenCreateTransactionBody() {
```

Comment:
This should be in `SyntheticTxFactory`
I think we can do it like this:


		@Override
		public TransactionBody.Builder body(final Bytes input, UnaryOperator<byte[]> aliasResolver) {
			tokenCreateOp = decoder.decodeFungibleCreate(input, aliasResolver);
                         verifyKeys(tokenCreateOp);
                         addInheritedKeys(tokenCreateOp); //or inline
			return   syntheticFactory.createCreateTokenSynthTx(tokenCreateOp);
		}